scripts/install-worker.sh

```diff
@@ -335,6 +365,64 @@ if [[ ! $KUBERNETES_VERSION =~ "1.19"* && ! $KUBERNETES_VERSION =~ "1.20"* && !
     sudo mv $TEMPLATE_DIR/ecr-credential-provider-config /etc/eks/ecr-credential-provider/ecr-credential-provider-config
 fi
 
+################################################################################
+### Cache Images ###############################################################
+################################################################################
+if [[ "$CACHE_CONTAINER_IMAGES" = "true" && "$BINARY_BUCKET_REGION" != "us-iso-east-1" && "$BINARY_BUCKET_REGION" != "us-isob-east-1" ]]; then
+    AWS_DOMAIN=$(imds 'latest/meta-data/services/domain')
+    ECR_URI=$(/etc/eks/get-ecr-uri.sh "${BINARY_BUCKET_REGION}" "${AWS_DOMAIN}")
+
+    PAUSE_CONTAINER="${ECR_URI}/eks/pause:${PAUSE_CONTAINER_VERSION}"
+    KUBE_PROXY_IMG="${ECR_URI}/eks/kube-proxy:v${KUBERNETES_VERSION}-${KUBE_PROXY_VERSION_SUFFIX}"
+    KUBE_PROXY_MINIMAL_IMG="${ECR_URI}/eks/kube-proxy:v${KUBERNETES_VERSION}-minimal-${KUBE_PROXY_VERSION_SUFFIX}"
+    cat /etc/eks/containerd/containerd-config.toml | sed s,SANDBOX_IMAGE,$PAUSE_CONTAINER,g | sudo tee /etc/eks/containerd/containerd-cached-pause-config.toml
+    sudo cp -v /etc/eks/containerd/containerd-cached-pause-config.toml /etc/containerd/config.toml
+
+    sudo cp -v /etc/eks/containerd/sandbox-image.service /etc/systemd/system/sandbox-image.service
+    sudo chown root:root /etc/systemd/system/sandbox-image.service
+    sudo systemctl daemon-reload
+    sudo systemctl start containerd
+    sudo systemctl enable containerd sandbox-image
+    /etc/eks/containerd/pull-image.sh "${PAUSE_CONTAINER}"
+    /etc/eks/containerd/pull-image.sh "${KUBE_PROXY_IMG}"
+    /etc/eks/containerd/pull-image.sh "${KUBE_PROXY_MINIMAL_IMG}"
+
+    #### Cache VPC CNI images starting with the addon default version and the latest version
+    K8S_MINOR_VERSION=$(echo "${KUBERNETES_VERSION}" | cut -d'.' -f1-2)
+    ADDON_VERSIONS=$(aws eks describe-addon-versions --addon-name vpc-cni --kubernetes-version=${K8S_MINOR_VERSION})
+    DEFAULT_VPC_CNI_VERSION=$(echo "${ADDON_VERSIONS}" | jq -r '.addons[] .addonVersions[] | select(.compatibilities[] .defaultVersion==true).addonVersion')
+    LATEST_VPC_CNI_VERSION=$(echo "${ADDON_VERSIONS}" | jq -r '.addons[] .addonVersions[] .addonVersion' | sort -V | tail -n1)
+    CNI_IMG="${ECR_URI}/amazon-k8s-cni"
+    CNI_INIT_IMG="${CNI_IMG}-init"
+    /etc/eks/containerd/pull-image.sh "${CNI_IMG}:${DEFAULT_VPC_CNI_VERSION}"
+    /etc/eks/containerd/pull-image.sh "${CNI_INIT_IMG}:${DEFAULT_VPC_CNI_VERSION}"
+    /etc/eks/containerd/pull-image.sh "${CNI_IMG}:${LATEST_VPC_CNI_VERSION}"
+    /etc/eks/containerd/pull-image.sh "${CNI_INIT_IMG}:${LATEST_VPC_CNI_VERSION}"
+    CNI_IMGS=("${CNI_IMG}:${DEFAULT_VPC_CNI_VERSION}" "${CNI_INIT_IMG}:${DEFAULT_VPC_CNI_VERSION}")
+    CNI_IMGS+=("${CNI_IMG}:${LATEST_VPC_CNI_VERSION}" "${CNI_INIT_IMG}:${LATEST_VPC_CNI_VERSION}")
```

Comment:
How about format like this?

```
CNI_IMGS=(
  # we could put some comment here
  "${CNI_IMG}:${DEFAULT_VPC_CNI_VERSION}"
  "${CNI_IMG}:${LATEST_VPC_CNI_VERSION}"

  # we could put some comment here
  "${CNI_INIT_IMG}:${DEFAULT_VPC_CNI_VERSION}"
  "${CNI_INIT_IMG}:${LATEST_VPC_CNI_VERSION}"
)
```

I think it would be easier to maintain if there have additional images add later on ( minimize diff line at next PR )

Reply:
that is a lot more readable :)